pkg/execute.go

```diff
@@ -161,3 +186,51 @@ func (e *Executor) execute(repo Repo, vaultClient *vault.Client, dryRun bool) er
 
 	return nil
 }
+
+// clones the output repo, writes the raw state to a file, commits and pushes that to GitLab
+func (e *Executor) commitAndPushState(repo Repo, state string) error {
+	tmpdir, err := os.MkdirTemp("", "tf-repo-state")
+	if err != nil {
+		return err
+	}
+	defer os.RemoveAll(tmpdir)
+
+	// clone with token https://docs.gitlab.com/ee/topics/git/clone.html#clone-using-a-token
+	args := []string{"-c", fmt.Sprintf(
+		"git clone https://%s:%s@%s .",
+		e.gitlabUsername,
+		e.gitlabToken,
+		e.gitlabLogRepo,
+	)}
+
+	_, err = executeCommand(tmpdir, "/bin/sh", args)
+	if err != nil {
+		return err
+	}
+
+	// prepare to template a markdown file
+	stateVars := &StateVars{
+		RepoName: repo.Name,
+		RepoURL:  repo.URL,
+		RepoSHA:  repo.Ref,
+		State:    state,
+	}
+
+	err = WriteTemplate(*stateVars, tmplData, fmt.Sprintf("%s/%s.md", tmpdir, repo.Name))
+
+	if err != nil {
+		return err
+	}
+
+	// commit and push the results
+	args = []string{"-c", fmt.Sprintf(
+		"git config user.email %s && git config user.name %s && git add %s.md && git commit -m '%s update' && git push origin",
+		e.gitEmail,
+		e.gitlabUsername,
+		repo.Name,
```

Comment:
Does this have to be a command?  No way to do this with go-git or similar?

Reply:
No I could definitely do it with go-git or another native golang git implementation. I just chose this way as the repo has already been setup to wrap git commands. I'm not opposed to trying a native implementation and cutting out the shell commands.